// @aws-sdk/client-rekognition/dist-es/commands/DeleteFacesCommand.js
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteFacesRequestFilterSensitiveLog, DeleteFacesResponseFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_json1_1DeleteFacesCommand, serializeAws_json1_1DeleteFacesCommand, } from "../protocols/Aws_json1_1";
export class DeleteFacesCommand extends $Command {
    constructor(input) {
        super();
        this.input = input;
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "RekognitionClient";
        const commandName = "DeleteFacesCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: DeleteFacesRequestFilterSensitiveLog,
            outputFilterSensitiveLog: DeleteFacesResponseFilterSensitiveLog,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return serializeAws_json1_1DeleteFacesCommand(input, context);
    }
    deserialize(output, context) {
        return deserializeAws_json1_1DeleteFacesCommand(output, context);
    }
}
