// @aws-sdk/node-config-provider/dist/cjs/fromSharedConfigFiles.spec.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var property_provider_1 = require("@aws-sdk/property-provider");
var shared_ini_file_loader_1 = require("@aws-sdk/shared-ini-file-loader");
var fromSharedConfigFiles_1 = require("./fromSharedConfigFiles");
jest.mock("@aws-sdk/shared-ini-file-loader", function () { return ({
    loadSharedConfigFiles: jest.fn(),
}); });
describe("fromSharedConfigFiles", function () {
    var envProfile = process.env[fromSharedConfigFiles_1.ENV_PROFILE];
    var configKey = "config_key";
    var configGetter = function (profile) { return profile[configKey]; };
    beforeEach(function () {
        delete process.env[fromSharedConfigFiles_1.ENV_PROFILE];
    });
    afterAll(function () {
        process.env[fromSharedConfigFiles_1.ENV_PROFILE] = envProfile;
    });
    var getProviderError = function (profile, getter) {
        return new property_provider_1.ProviderError("Cannot load config for profile " + profile + " in SDK configuration files with getter: " + getter);
    };
    describe("loadedConfig", function () {
        var _a, _b, _c, _d, _e, _f, _g, _h, _j, _k, _l, _m;
        var mockConfigAnswer = "mockConfigAnswer";
        var mockConfigNotAnswer = "mockConfigNotAnswer";
        var mockCredentialsAnswer = "mockCredentialsAnswer";
        var mockCredentialsNotAnswer = "mockCredentialsNotAnswer";
        var loadedConfigResolves = [
            {
                message: "returns configValue from default profile",
                iniDataInConfig: {
                    default: (_a = {}, _a[configKey] = mockConfigAnswer, _a),
                },
                iniDataInCredentials: {
                    default: (_b = {}, _b[configKey] = mockCredentialsNotAnswer, _b),
                },
                configValueToVerify: mockConfigAnswer,
            },
            {
                message: "returns configValue from designated profile",
                iniDataInConfig: {
                    default: (_c = {}, _c[configKey] = mockConfigNotAnswer, _c),
                    foo: (_d = {}, _d[configKey] = mockConfigAnswer, _d),
                },
                iniDataInCredentials: {
                    foo: (_e = {}, _e[configKey] = mockCredentialsNotAnswer, _e),
                },
                profile: "foo",
                configValueToVerify: mockConfigAnswer,
            },
            {
                message: "returns configValue from credentials file if preferred",
                iniDataInConfig: {
                    default: (_f = {}, _f[configKey] = mockConfigNotAnswer, _f),
                    foo: (_g = {}, _g[configKey] = mockConfigNotAnswer, _g),
                },
                iniDataInCredentials: {
                    foo: (_h = {}, _h[configKey] = mockCredentialsAnswer, _h),
                },
                profile: "foo",
                preferredFile: "credentials",
                configValueToVerify: mockCredentialsAnswer,
            },
            {
                message: "returns configValue from config file if preferred credentials file doesn't contain config",
                iniDataInConfig: {
                    foo: (_j = {}, _j[configKey] = mockConfigAnswer, _j),
                },
                iniDataInCredentials: {},
                configValueToVerify: mockConfigAnswer,
                preferredFile: "credentials",
                profile: "foo",
            },
            {
                message: "returns configValue from credential file if preferred config file doesn't contain config",
                iniDataInConfig: {},
                iniDataInCredentials: {
                    foo: (_k = {}, _k[configKey] = mockCredentialsAnswer, _k),
                },
                configValueToVerify: mockCredentialsAnswer,
                profile: "foo",
            },
        ];
        var loadedConfigRejects = [
            {
                message: "rejects if default profile is not present and profile value is not passed",
                iniDataInConfig: {
                    foo: (_l = {}, _l[configKey] = mockConfigNotAnswer, _l),
                },
                iniDataInCredentials: {},
            },
            {
                message: "rejects if designated profile is not present",
                iniDataInConfig: {
                    default: (_m = {}, _m[configKey] = mockConfigNotAnswer, _m),
                },
                iniDataInCredentials: {},
                profile: "foo",
            },
        ];
        describe("uses the shared ini file loader if pre-loaded config is not supplied", function () {
            loadedConfigResolves.forEach(function (_a) {
                var message = _a.message, iniDataInConfig = _a.iniDataInConfig, iniDataInCredentials = _a.iniDataInCredentials, configValueToVerify = _a.configValueToVerify, profile = _a.profile, preferredFile = _a.preferredFile;
                it(message, function () {
                    shared_ini_file_loader_1.loadSharedConfigFiles.mockResolvedValueOnce({
                        configFile: iniDataInConfig,
                        credentialsFile: iniDataInCredentials,
                    });
                    return expect(fromSharedConfigFiles_1.fromSharedConfigFiles(configGetter, { profile: profile, preferredFile: preferredFile })()).resolves.toBe(configValueToVerify);
                });
            });
            loadedConfigRejects.forEach(function (_a) {
                var message = _a.message, iniDataInConfig = _a.iniDataInConfig, iniDataInCredentials = _a.iniDataInCredentials, profile = _a.profile, preferredFile = _a.preferredFile;
                it(message, function () {
                    shared_ini_file_loader_1.loadSharedConfigFiles.mockResolvedValueOnce({
                        configFile: iniDataInConfig,
                        credentialsFile: iniDataInCredentials,
                    });
                    return expect(fromSharedConfigFiles_1.fromSharedConfigFiles(configGetter, { profile: profile, preferredFile: preferredFile })()).rejects.toMatchObject(getProviderError(profile !== null && profile !== void 0 ? profile : "default", configGetter));
                });
            });
        });
        describe("uses pre-loaded config if supplied", function () {
            loadedConfigResolves.forEach(function (_a) {
                var message = _a.message, iniDataInConfig = _a.iniDataInConfig, iniDataInCredentials = _a.iniDataInCredentials, configValueToVerify = _a.configValueToVerify, profile = _a.profile, preferredFile = _a.preferredFile;
                it(message + " from config file", function () {
                    var loadedConfig = Promise.resolve({
                        configFile: iniDataInConfig,
                        credentialsFile: iniDataInCredentials,
                    });
                    return expect(fromSharedConfigFiles_1.fromSharedConfigFiles(configGetter, { loadedConfig: loadedConfig, profile: profile, preferredFile: preferredFile })()).resolves.toBe(configValueToVerify);
                });
            });
            loadedConfigRejects.forEach(function (_a) {
                var message = _a.message, iniDataInConfig = _a.iniDataInConfig, iniDataInCredentials = _a.iniDataInCredentials, profile = _a.profile, preferredFile = _a.preferredFile;
                it(message, function () {
                    var loadedConfig = Promise.resolve({
                        configFile: iniDataInConfig,
                        credentialsFile: iniDataInCredentials,
                    });
                    return expect(fromSharedConfigFiles_1.fromSharedConfigFiles(configGetter, { loadedConfig: loadedConfig, profile: profile, preferredFile: preferredFile })()).rejects.toMatchObject(getProviderError(profile !== null && profile !== void 0 ? profile : "default", configGetter));
                });
            });
        });
        it("rejects if getter throws", function () {
            var message = "Cannot load config";
            var failGetter = function () {
                throw new Error(message);
            };
            shared_ini_file_loader_1.loadSharedConfigFiles.mockResolvedValueOnce({
                configFile: {},
                credentialsFile: {},
            });
            return expect(fromSharedConfigFiles_1.fromSharedConfigFiles(failGetter)()).rejects.toMatchObject(new property_provider_1.ProviderError(message));
        });
    });
    describe("profile", function () {
        var _a, _b, _c;
        var loadedConfigData = {
            configFile: {
                default: (_a = {}, _a[configKey] = "configFileDefault", _a),
                foo: (_b = {}, _b[configKey] = "configFileFoo", _b),
            },
            credentialsFile: {
                default: (_c = {}, _c[configKey] = "credentialsFileDefault", _c),
            },
        };
        var loadedConfig = Promise.resolve(loadedConfigData);
        describe("when profile is not defined", function () {
            it("returns configValue from value in '" + fromSharedConfigFiles_1.ENV_PROFILE + "' env var if it is set", function () {
                var profile = "foo";
                process.env[fromSharedConfigFiles_1.ENV_PROFILE] = profile;
                return expect(fromSharedConfigFiles_1.fromSharedConfigFiles(configGetter, { loadedConfig: loadedConfig })()).resolves.toBe(loadedConfigData.configFile[profile][configKey]);
            });
            it("returns configValue from default profile if '" + fromSharedConfigFiles_1.ENV_PROFILE + "' env var is not set", function () {
                return expect(fromSharedConfigFiles_1.fromSharedConfigFiles(configGetter, { loadedConfig: loadedConfig })()).resolves.toBe(loadedConfigData.configFile.default[configKey]);
            });
        });
    });
});
