// @aws-sdk/client-rekognition/dist-cjs/models/models_0.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.S3ObjectFilterSensitiveLog = exports.AgeRangeFilterSensitiveLog = exports.StreamProcessorParameterToDelete = exports.VideoTooLargeException = exports.MalformedPolicyDocumentException = exports.Reason = exports.IdempotentParameterMismatchException = exports.SegmentType = exports.TechnicalCueType = exports.PersonTrackingSortBy = exports.LabelDetectionSortBy = exports.VideoColorRange = exports.VideoJobStatus = exports.FaceSearchSortBy = exports.FaceAttributes = exports.TextTypes = exports.HumanLoopQuotaExceededException = exports.ResourceNotReadyException = exports.StreamProcessorStatus = exports.InvalidPaginationTokenException = exports.ProjectVersionStatus = exports.InvalidPolicyRevisionIdException = exports.ProjectStatus = exports.DatasetStatusMessageCode = exports.DatasetStatus = exports.DatasetType = exports.ResourceAlreadyExistsException = exports.ServiceQuotaExceededException = exports.ResourceNotFoundException = exports.ResourceInUseException = exports.LimitExceededException = exports.ContentModerationSortBy = exports.ContentClassifier = exports.ThrottlingException = exports.ProvisionedThroughputExceededException = exports.InvalidS3ObjectException = exports.InvalidParameterException = exports.InvalidImageFormatException = exports.InternalServerError = exports.ImageTooLargeException = exports.OrientationCorrection = exports.QualityFilter = exports.CelebrityRecognitionSortBy = exports.GenderType = exports.KnownGenderType = exports.LandmarkType = exports.ProtectiveEquipmentType = exports.BodyPart = exports.Attribute = exports.AccessDeniedException = void 0;
exports.StreamProcessorDataSharingPreferenceFilterSensitiveLog = exports.CreateProjectVersionResponseFilterSensitiveLog = exports.CreateProjectVersionRequestFilterSensitiveLog = exports.TrainingDataFilterSensitiveLog = exports.TestingDataFilterSensitiveLog = exports.CreateProjectResponseFilterSensitiveLog = exports.CreateProjectRequestFilterSensitiveLog = exports.CreateDatasetResponseFilterSensitiveLog = exports.CreateDatasetRequestFilterSensitiveLog = exports.DatasetSourceFilterSensitiveLog = exports.CreateCollectionResponseFilterSensitiveLog = exports.CreateCollectionRequestFilterSensitiveLog = exports.CopyProjectVersionResponseFilterSensitiveLog = exports.CopyProjectVersionRequestFilterSensitiveLog = exports.OutputConfigFilterSensitiveLog = exports.ContentModerationDetectionFilterSensitiveLog = exports.ModerationLabelFilterSensitiveLog = exports.ConnectedHomeSettingsForUpdateFilterSensitiveLog = exports.ConnectedHomeSettingsFilterSensitiveLog = exports.CompareFacesResponseFilterSensitiveLog = exports.CompareFacesMatchFilterSensitiveLog = exports.CompareFacesRequestFilterSensitiveLog = exports.ImageFilterSensitiveLog = exports.ComparedSourceImageFaceFilterSensitiveLog = exports.CelebrityRecognitionFilterSensitiveLog = exports.CelebrityDetailFilterSensitiveLog = exports.FaceDetailFilterSensitiveLog = exports.SunglassesFilterSensitiveLog = exports.MustacheFilterSensitiveLog = exports.MouthOpenFilterSensitiveLog = exports.GenderFilterSensitiveLog = exports.EyeOpenFilterSensitiveLog = exports.EyeglassesFilterSensitiveLog = exports.CelebrityFilterSensitiveLog = exports.KnownGenderFilterSensitiveLog = exports.ComparedFaceFilterSensitiveLog = exports.SmileFilterSensitiveLog = exports.ImageQualityFilterSensitiveLog = exports.PoseFilterSensitiveLog = exports.LandmarkFilterSensitiveLog = exports.EmotionFilterSensitiveLog = exports.ProtectiveEquipmentBodyPartFilterSensitiveLog = exports.EquipmentDetectionFilterSensitiveLog = exports.CoversBodyPartFilterSensitiveLog = exports.BoundingBoxFilterSensitiveLog = exports.BlackFrameFilterSensitiveLog = exports.BeardFilterSensitiveLog = exports.AudioMetadataFilterSensitiveLog = exports.AssetFilterSensitiveLog = exports.GroundTruthManifestFilterSensitiveLog = void 0;
exports.DescribeStreamProcessorRequestFilterSensitiveLog = exports.DescribeProjectVersionsResponseFilterSensitiveLog = exports.ProjectVersionDescriptionFilterSensitiveLog = exports.TrainingDataResultFilterSensitiveLog = exports.TestingDataResultFilterSensitiveLog = exports.ValidationDataFilterSensitiveLog = exports.EvaluationResultFilterSensitiveLog = exports.SummaryFilterSensitiveLog = exports.DescribeProjectVersionsRequestFilterSensitiveLog = exports.DescribeProjectsResponseFilterSensitiveLog = exports.ProjectDescriptionFilterSensitiveLog = exports.DescribeProjectsRequestFilterSensitiveLog = exports.DescribeDatasetResponseFilterSensitiveLog = exports.DescribeDatasetRequestFilterSensitiveLog = exports.DescribeCollectionResponseFilterSensitiveLog = exports.DescribeCollectionRequestFilterSensitiveLog = exports.DeleteStreamProcessorResponseFilterSensitiveLog = exports.DeleteStreamProcessorRequestFilterSensitiveLog = exports.DeleteProjectVersionResponseFilterSensitiveLog = exports.DeleteProjectVersionRequestFilterSensitiveLog = exports.DeleteProjectPolicyResponseFilterSensitiveLog = exports.DeleteProjectPolicyRequestFilterSensitiveLog = exports.DeleteProjectResponseFilterSensitiveLog = exports.DeleteProjectRequestFilterSensitiveLog = exports.DeleteFacesResponseFilterSensitiveLog = exports.DeleteFacesRequestFilterSensitiveLog = exports.DeleteDatasetResponseFilterSensitiveLog = exports.DeleteDatasetRequestFilterSensitiveLog = exports.DeleteCollectionResponseFilterSensitiveLog = exports.DeleteCollectionRequestFilterSensitiveLog = exports.DatasetMetadataFilterSensitiveLog = exports.DatasetLabelDescriptionFilterSensitiveLog = exports.DatasetLabelStatsFilterSensitiveLog = exports.DatasetDescriptionFilterSensitiveLog = exports.DatasetStatsFilterSensitiveLog = exports.DatasetChangesFilterSensitiveLog = exports.CustomLabelFilterSensitiveLog = exports.GeometryFilterSensitiveLog = exports.CreateStreamProcessorResponseFilterSensitiveLog = exports.CreateStreamProcessorRequestFilterSensitiveLog = exports.StreamProcessorSettingsFilterSensitiveLog = exports.FaceSearchSettingsFilterSensitiveLog = exports.RegionOfInterestFilterSensitiveLog = exports.PointFilterSensitiveLog = exports.StreamProcessorOutputFilterSensitiveLog = exports.S3DestinationFilterSensitiveLog = exports.KinesisDataStreamFilterSensitiveLog = exports.StreamProcessorNotificationChannelFilterSensitiveLog = exports.StreamProcessorInputFilterSensitiveLog = exports.KinesisVideoStreamFilterSensitiveLog = void 0;
exports.PersonDetectionFilterSensitiveLog = exports.GetPersonTrackingRequestFilterSensitiveLog = exports.GetLabelDetectionResponseFilterSensitiveLog = exports.LabelDetectionFilterSensitiveLog = exports.GetLabelDetectionRequestFilterSensitiveLog = exports.GetFaceSearchResponseFilterSensitiveLog = exports.PersonMatchFilterSensitiveLog = exports.PersonDetailFilterSensitiveLog = exports.GetFaceSearchRequestFilterSensitiveLog = exports.GetFaceDetectionResponseFilterSensitiveLog = exports.GetFaceDetectionRequestFilterSensitiveLog = exports.GetContentModerationResponseFilterSensitiveLog = exports.GetContentModerationRequestFilterSensitiveLog = exports.GetCelebrityRecognitionResponseFilterSensitiveLog = exports.VideoMetadataFilterSensitiveLog = exports.GetCelebrityRecognitionRequestFilterSensitiveLog = exports.GetCelebrityInfoResponseFilterSensitiveLog = exports.GetCelebrityInfoRequestFilterSensitiveLog = exports.FaceRecordFilterSensitiveLog = exports.FaceMatchFilterSensitiveLog = exports.FaceDetectionFilterSensitiveLog = exports.FaceFilterSensitiveLog = exports.DistributeDatasetEntriesResponseFilterSensitiveLog = exports.DistributeDatasetEntriesRequestFilterSensitiveLog = exports.DistributeDatasetFilterSensitiveLog = exports.DetectTextResponseFilterSensitiveLog = exports.TextDetectionFilterSensitiveLog = exports.DetectTextRequestFilterSensitiveLog = exports.DetectTextFiltersFilterSensitiveLog = exports.DetectProtectiveEquipmentResponseFilterSensitiveLog = exports.ProtectiveEquipmentSummaryFilterSensitiveLog = exports.ProtectiveEquipmentPersonFilterSensitiveLog = exports.DetectProtectiveEquipmentRequestFilterSensitiveLog = exports.ProtectiveEquipmentSummarizationAttributesFilterSensitiveLog = exports.DetectModerationLabelsResponseFilterSensitiveLog = exports.HumanLoopActivationOutputFilterSensitiveLog = exports.DetectModerationLabelsRequestFilterSensitiveLog = exports.HumanLoopConfigFilterSensitiveLog = exports.HumanLoopDataAttributesFilterSensitiveLog = exports.DetectLabelsResponseFilterSensitiveLog = exports.LabelFilterSensitiveLog = exports.ParentFilterSensitiveLog = exports.InstanceFilterSensitiveLog = exports.DetectLabelsRequestFilterSensitiveLog = exports.DetectionFilterFilterSensitiveLog = exports.DetectFacesResponseFilterSensitiveLog = exports.DetectFacesRequestFilterSensitiveLog = exports.DetectCustomLabelsResponseFilterSensitiveLog = exports.DetectCustomLabelsRequestFilterSensitiveLog = exports.DescribeStreamProcessorResponseFilterSensitiveLog = void 0;
exports.StartLabelDetectionResponseFilterSensitiveLog = exports.StartLabelDetectionRequestFilterSensitiveLog = exports.StartFaceSearchResponseFilterSensitiveLog = exports.StartFaceSearchRequestFilterSensitiveLog = exports.StartFaceDetectionResponseFilterSensitiveLog = exports.StartFaceDetectionRequestFilterSensitiveLog = exports.StartContentModerationResponseFilterSensitiveLog = exports.StartContentModerationRequestFilterSensitiveLog = exports.StartCelebrityRecognitionResponseFilterSensitiveLog = exports.StartCelebrityRecognitionRequestFilterSensitiveLog = exports.VideoFilterSensitiveLog = exports.SearchFacesByImageResponseFilterSensitiveLog = exports.SearchFacesByImageRequestFilterSensitiveLog = exports.SearchFacesResponseFilterSensitiveLog = exports.SearchFacesRequestFilterSensitiveLog = exports.RecognizeCelebritiesResponseFilterSensitiveLog = exports.RecognizeCelebritiesRequestFilterSensitiveLog = exports.PutProjectPolicyResponseFilterSensitiveLog = exports.PutProjectPolicyRequestFilterSensitiveLog = exports.NotificationChannelFilterSensitiveLog = exports.ListTagsForResourceResponseFilterSensitiveLog = exports.ListTagsForResourceRequestFilterSensitiveLog = exports.ListStreamProcessorsResponseFilterSensitiveLog = exports.StreamProcessorFilterSensitiveLog = exports.ListStreamProcessorsRequestFilterSensitiveLog = exports.ListProjectPoliciesResponseFilterSensitiveLog = exports.ProjectPolicyFilterSensitiveLog = exports.ListProjectPoliciesRequestFilterSensitiveLog = exports.ListFacesResponseFilterSensitiveLog = exports.ListFacesRequestFilterSensitiveLog = exports.ListDatasetLabelsResponseFilterSensitiveLog = exports.ListDatasetLabelsRequestFilterSensitiveLog = exports.ListDatasetEntriesResponseFilterSensitiveLog = exports.ListDatasetEntriesRequestFilterSensitiveLog = exports.ListCollectionsResponseFilterSensitiveLog = exports.ListCollectionsRequestFilterSensitiveLog = exports.KinesisVideoStreamStartSelectorFilterSensitiveLog = exports.IndexFacesResponseFilterSensitiveLog = exports.UnindexedFaceFilterSensitiveLog = exports.IndexFacesRequestFilterSensitiveLog = exports.GetTextDetectionResponseFilterSensitiveLog = exports.TextDetectionResultFilterSensitiveLog = exports.GetTextDetectionRequestFilterSensitiveLog = exports.GetSegmentDetectionResponseFilterSensitiveLog = exports.SegmentTypeInfoFilterSensitiveLog = exports.SegmentDetectionFilterSensitiveLog = exports.TechnicalCueSegmentFilterSensitiveLog = exports.ShotSegmentFilterSensitiveLog = exports.GetSegmentDetectionRequestFilterSensitiveLog = exports.GetPersonTrackingResponseFilterSensitiveLog = void 0;
exports.UpdateStreamProcessorResponseFilterSensitiveLog = exports.UpdateStreamProcessorRequestFilterSensitiveLog = exports.StreamProcessorSettingsForUpdateFilterSensitiveLog = exports.UpdateDatasetEntriesResponseFilterSensitiveLog = exports.UpdateDatasetEntriesRequestFilterSensitiveLog = exports.UntagResourceResponseFilterSensitiveLog = exports.UntagResourceRequestFilterSensitiveLog = exports.TagResourceResponseFilterSensitiveLog = exports.TagResourceRequestFilterSensitiveLog = exports.StopStreamProcessorResponseFilterSensitiveLog = exports.StopStreamProcessorRequestFilterSensitiveLog = exports.StopProjectVersionResponseFilterSensitiveLog = exports.StopProjectVersionRequestFilterSensitiveLog = exports.StartTextDetectionResponseFilterSensitiveLog = exports.StartTextDetectionRequestFilterSensitiveLog = exports.StartTextDetectionFiltersFilterSensitiveLog = exports.StartStreamProcessorResponseFilterSensitiveLog = exports.StartStreamProcessorRequestFilterSensitiveLog = exports.StreamProcessingStopSelectorFilterSensitiveLog = exports.StreamProcessingStartSelectorFilterSensitiveLog = exports.StartSegmentDetectionResponseFilterSensitiveLog = exports.StartSegmentDetectionRequestFilterSensitiveLog = exports.StartSegmentDetectionFiltersFilterSensitiveLog = exports.StartTechnicalCueDetectionFilterFilterSensitiveLog = exports.StartShotDetectionFilterFilterSensitiveLog = exports.StartProjectVersionResponseFilterSensitiveLog = exports.StartProjectVersionRequestFilterSensitiveLog = exports.StartPersonTrackingResponseFilterSensitiveLog = exports.StartPersonTrackingRequestFilterSensitiveLog = void 0;
const RekognitionServiceException_1 = require("./RekognitionServiceException");
class AccessDeniedException extends RekognitionServiceException_1.RekognitionServiceException {
    constructor(opts) {
        super({
            name: "AccessDeniedException",
            $fault: "client",
            ...opts,
        });
        this.name = "AccessDeniedException";
        this.$fault = "client";
        Object.setPrototypeOf(this, AccessDeniedException.prototype);
        this.Message = opts.Message;
        this.Code = opts.Code;
        this.Logref = opts.Logref;
    }
}
exports.AccessDeniedException = AccessDeniedException;
var Attribute;
(function (Attribute) {
    Attribute["ALL"] = "ALL";
    Attribute["DEFAULT"] = "DEFAULT";
})(Attribute = exports.Attribute || (exports.Attribute = {}));
var BodyPart;
(function (BodyPart) {
    BodyPart["FACE"] = "FACE";
    BodyPart["HEAD"] = "HEAD";
    BodyPart["LEFT_HAND"] = "LEFT_HAND";
    BodyPart["RIGHT_HAND"] = "RIGHT_HAND";
})(BodyPart = exports.BodyPart || (exports.BodyPart = {}));
var ProtectiveEquipmentType;
(function (ProtectiveEquipmentType) {
    ProtectiveEquipmentType["FACE_COVER"] = "FACE_COVER";
    ProtectiveEquipmentType["HAND_COVER"] = "HAND_COVER";
    ProtectiveEquipmentType["HEAD_COVER"] = "HEAD_COVER";
})(ProtectiveEquipmentType = exports.ProtectiveEquipmentType || (exports.ProtectiveEquipmentType = {}));
var LandmarkType;
(function (LandmarkType) {
    LandmarkType["chinBottom"] = "chinBottom";
    LandmarkType["eyeLeft"] = "eyeLeft";
    LandmarkType["eyeRight"] = "eyeRight";
    LandmarkType["leftEyeBrowLeft"] = "leftEyeBrowLeft";
    LandmarkType["leftEyeBrowRight"] = "leftEyeBrowRight";
    LandmarkType["leftEyeBrowUp"] = "leftEyeBrowUp";
    LandmarkType["leftEyeDown"] = "leftEyeDown";
    LandmarkType["leftEyeLeft"] = "leftEyeLeft";
    LandmarkType["leftEyeRight"] = "leftEyeRight";
    LandmarkType["leftEyeUp"] = "leftEyeUp";
    LandmarkType["leftPupil"] = "leftPupil";
    LandmarkType["midJawlineLeft"] = "midJawlineLeft";
    LandmarkType["midJawlineRight"] = "midJawlineRight";
    LandmarkType["mouthDown"] = "mouthDown";
    LandmarkType["mouthLeft"] = "mouthLeft";
    LandmarkType["mouthRight"] = "mouthRight";
    LandmarkType["mouthUp"] = "mouthUp";
    LandmarkType["nose"] = "nose";
    LandmarkType["noseLeft"] = "noseLeft";
    LandmarkType["noseRight"] = "noseRight";
    LandmarkType["rightEyeBrowLeft"] = "rightEyeBrowLeft";
    LandmarkType["rightEyeBrowRight"] = "rightEyeBrowRight";
    LandmarkType["rightEyeBrowUp"] = "rightEyeBrowUp";
    LandmarkType["rightEyeDown"] = "rightEyeDown";
    LandmarkType["rightEyeLeft"] = "rightEyeLeft";
    LandmarkType["rightEyeRight"] = "rightEyeRight";
    LandmarkType["rightEyeUp"] = "rightEyeUp";
    LandmarkType["rightPupil"] = "rightPupil";
    LandmarkType["upperJawlineLeft"] = "upperJawlineLeft";
    LandmarkType["upperJawlineRight"] = "upperJawlineRight";
})(LandmarkType = exports.LandmarkType || (exports.LandmarkType = {}));
var KnownGenderType;
(function (KnownGenderType) {
    KnownGenderType["Female"] = "Female";
    KnownGenderType["Male"] = "Male";
    KnownGenderType["Nonbinary"] = "Nonbinary";
    KnownGenderType["Unlisted"] = "Unlisted";
})(KnownGenderType = exports.KnownGenderType || (exports.KnownGenderType = {}));
var GenderType;
(function (GenderType) {
    GenderType["Female"] = "Female";
    GenderType["Male"] = "Male";
})(GenderType = exports.GenderType || (exports.GenderType = {}));
var CelebrityRecognitionSortBy;
(function (CelebrityRecognitionSortBy) {
    CelebrityRecognitionSortBy["ID"] = "ID";
    CelebrityRecognitionSortBy["TIMESTAMP"] = "TIMESTAMP";
})(CelebrityRecognitionSortBy = exports.CelebrityRecognitionSortBy || (exports.CelebrityRecognitionSortBy = {}));
var QualityFilter;
(function (QualityFilter) {
    QualityFilter["AUTO"] = "AUTO";
    QualityFilter["HIGH"] = "HIGH";
    QualityFilter["LOW"] = "LOW";
    QualityFilter["MEDIUM"] = "MEDIUM";
    QualityFilter["NONE"] = "NONE";
})(QualityFilter = exports.QualityFilter || (exports.QualityFilter = {}));
var OrientationCorrection;
(function (OrientationCorrection) {
    OrientationCorrection["ROTATE_0"] = "ROTATE_0";
    OrientationCorrection["ROTATE_180"] = "ROTATE_180";
    OrientationCorrection["ROTATE_270"] = "ROTATE_270";
    OrientationCorrection["ROTATE_90"] = "ROTATE_90";
})(OrientationCorrection = exports.OrientationCorrection || (exports.OrientationCorrection = {}));
class ImageTooLargeException extends RekognitionServiceException_1.RekognitionServiceException {
    constructor(opts) {
        super({
            name: "ImageTooLargeException",
            $fault: "client",
            ...opts,
        });
        this.name = "ImageTooLargeException";
        this.$fault = "client";
        Object.setPrototypeOf(this, ImageTooLargeException.prototype);
        this.Message = opts.Message;
        this.Code = opts.Code;
        this.Logref = opts.Logref;
    }
}
exports.ImageTooLargeException = ImageTooLargeException;
class InternalServerError extends RekognitionServiceException_1.RekognitionServiceException {
    constructor(opts) {
        super({
            name: "InternalServerError",
            $fault: "server",
            ...opts,
        });
        this.name = "InternalServerError";
        this.$fault = "server";
        Object.setPrototypeOf(this, InternalServerError.prototype);
        this.Message = opts.Message;
        this.Code = opts.Code;
        this.Logref = opts.Logref;
    }
}
exports.InternalServerError = InternalServerError;
class InvalidImageFormatException extends RekognitionServiceException_1.RekognitionServiceException {
    constructor(opts) {
        super({
            name: "InvalidImageFormatException",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidImageFormatException";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidImageFormatException.prototype);
        this.Message = opts.Message;
        this.Code = opts.Code;
        this.Logref = opts.Logref;
    }
}
exports.InvalidImageFormatException = InvalidImageFormatException;
class InvalidParameterException extends RekognitionServiceException_1.RekognitionServiceException {
    constructor(opts) {
        super({
            name: "InvalidParameterException",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidParameterException";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidParameterException.prototype);
        this.Message = opts.Message;
        this.Code = opts.Code;
        this.Logref = opts.Logref;
    }
}
exports.InvalidParameterException = InvalidParameterException;
class InvalidS3ObjectException extends RekognitionServiceException_1.RekognitionServiceException {
    constructor(opts) {
        super({
            name: "InvalidS3ObjectException",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidS3ObjectException";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidS3ObjectException.prototype);
        this.Message = opts.Message;
        this.Code = opts.Code;
        this.Logref = opts.Logref;
    }
}
exports.InvalidS3ObjectException = InvalidS3ObjectException;
class ProvisionedThroughputExceededException extends RekognitionServiceException_1.RekognitionServiceException {
    constructor(opts) {
        super({
            name: "ProvisionedThroughputExceededException",
            $fault: "client",
            ...opts,
        });
        this.name = "ProvisionedThroughputExceededException";
        this.$fault = "client";
        Object.setPrototypeOf(this, ProvisionedThroughputExceededException.prototype);
        this.Message = opts.Message;
        this.Code = opts.Code;
        this.Logref = opts.Logref;
    }
}
exports.ProvisionedThroughputExceededException = ProvisionedThroughputExceededException;
class ThrottlingException extends RekognitionServiceException_1.RekognitionServiceException {
    constructor(opts) {
        super({
            name: "ThrottlingException",
            $fault: "server",
            ...opts,
        });
        this.name = "ThrottlingException";
        this.$fault = "server";
        Object.setPrototypeOf(this, ThrottlingException.prototype);
        this.Message = opts.Message;
        this.Code = opts.Code;
        this.Logref = opts.Logref;
    }
}
exports.ThrottlingException = ThrottlingException;
var ContentClassifier;
(function (ContentClassifier) {
    ContentClassifier["FREE_OF_ADULT_CONTENT"] = "FreeOfAdultContent";
    ContentClassifier["FREE_OF_PERSONALLY_IDENTIFIABLE_INFORMATION"] = "FreeOfPersonallyIdentifiableInformation";
})(ContentClassifier = exports.ContentClassifier || (exports.ContentClassifier = {}));
var ContentModerationSortBy;
(function (ContentModerationSortBy) {
    ContentModerationSortBy["NAME"] = "NAME";
    ContentModerationSortBy["TIMESTAMP"] = "TIMESTAMP";
})(ContentModerationSortBy = exports.ContentModerationSortBy || (exports.ContentModerationSortBy = {}));
class LimitExceededException extends RekognitionServiceException_1.RekognitionServiceException {
    constructor(opts) {
        super({
            name: "LimitExceededException",
            $fault: "client",
            ...opts,
        });
        this.name = "LimitExceededException";
        this.$fault = "client";
        Object.setPrototypeOf(this, LimitExceededException.prototype);
        this.Message = opts.Message;
        this.Code = opts.Code;
        this.Logref = opts.Logref;
    }
}
exports.LimitExceededException = LimitExceededException;
class ResourceInUseException extends RekognitionServiceException_1.RekognitionServiceException {
    constructor(opts) {
        super({
            name: "ResourceInUseException",
            $fault: "client",
            ...opts,
        });
        this.name = "ResourceInUseException";
        this.$fault = "client";
        Object.setPrototypeOf(this, ResourceInUseException.prototype);
        this.Message = opts.Message;
        this.Code = opts.Code;
        this.Logref = opts.Logref;
    }
}
exports.ResourceInUseException = ResourceInUseException;
class ResourceNotFoundException extends RekognitionServiceException_1.RekognitionServiceException {
    constructor(opts) {
        super({
            name: "ResourceNotFoundException",
            $fault: "client",
            ...opts,
        });
        this.name = "ResourceNotFoundException";
        this.$fault = "client";
        Object.setPrototypeOf(this, ResourceNotFoundException.prototype);
        this.Message = opts.Message;
        this.Code = opts.Code;
        this.Logref = opts.Logref;
    }
}
exports.ResourceNotFoundException = ResourceNotFoundException;
class ServiceQuotaExceededException extends RekognitionServiceException_1.RekognitionServiceException {
    constructor(opts) {
        super({
            name: "ServiceQuotaExceededException",
            $fault: "client",
            ...opts,
        });
        this.name = "ServiceQuotaExceededException";
        this.$fault = "client";
        Object.setPrototypeOf(this, ServiceQuotaExceededException.prototype);
        this.Message = opts.Message;
        this.Code = opts.Code;
        this.Logref = opts.Logref;
    }
}
exports.ServiceQuotaExceededException = ServiceQuotaExceededException;
class ResourceAlreadyExistsException extends RekognitionServiceException_1.RekognitionServiceException {
    constructor(opts) {
        super({
            name: "ResourceAlreadyExistsException",
            $fault: "client",
            ...opts,
        });
        this.name = "ResourceAlreadyExistsException";
        this.$fault = "client";
        Object.setPrototypeOf(this, ResourceAlreadyExistsException.prototype);
        this.Message = opts.Message;
        this.Code = opts.Code;
        this.Logref = opts.Logref;
    }
}
exports.ResourceAlreadyExistsException = ResourceAlreadyExistsException;
var DatasetType;
(function (DatasetType) {
    DatasetType["TEST"] = "TEST";
    DatasetType["TRAIN"] = "TRAIN";
})(DatasetType = exports.DatasetType || (exports.DatasetType = {}));
var DatasetStatus;
(function (DatasetStatus) {
    DatasetStatus["CREATE_COMPLETE"] = "CREATE_COMPLETE";
    DatasetStatus["CREATE_FAILED"] = "CREATE_FAILED";
    DatasetStatus["CREATE_IN_PROGRESS"] = "CREATE_IN_PROGRESS";
    DatasetStatus["DELETE_IN_PROGRESS"] = "DELETE_IN_PROGRESS";
    DatasetStatus["UPDATE_COMPLETE"] = "UPDATE_COMPLETE";
    DatasetStatus["UPDATE_FAILED"] = "UPDATE_FAILED";
    DatasetStatus["UPDATE_IN_PROGRESS"] = "UPDATE_IN_PROGRESS";
})(DatasetStatus = exports.DatasetStatus || (exports.DatasetStatus = {}));
var DatasetStatusMessageCode;
(function (DatasetStatusMessageCode) {
    DatasetStatusMessageCode["CLIENT_ERROR"] = "CLIENT_ERROR";
    DatasetStatusMessageCode["SERVICE_ERROR"] = "SERVICE_ERROR";
    DatasetStatusMessageCode["SUCCESS"] = "SUCCESS";
})(DatasetStatusMessageCode = exports.DatasetStatusMessageCode || (exports.DatasetStatusMessageCode = {}));
var ProjectStatus;
(function (ProjectStatus) {
    ProjectStatus["CREATED"] = "CREATED";
    ProjectStatus["CREATING"] = "CREATING";
    ProjectStatus["DELETING"] = "DELETING";
})(ProjectStatus = exports.ProjectStatus || (exports.ProjectStatus = {}));
class InvalidPolicyRevisionIdException extends RekognitionServiceException_1.RekognitionServiceException {
    constructor(opts) {
        super({
            name: "InvalidPolicyRevisionIdException",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidPolicyRevisionIdException";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidPolicyRevisionIdException.prototype);
        this.Message = opts.Message;
        this.Code = opts.Code;
        this.Logref = opts.Logref;
    }
}
exports.InvalidPolicyRevisionIdException = InvalidPolicyRevisionIdException;
var ProjectVersionStatus;
(function (ProjectVersionStatus) {
    ProjectVersionStatus["COPYING_COMPLETED"] = "COPYING_COMPLETED";
    ProjectVersionStatus["COPYING_FAILED"] = "COPYING_FAILED";
    ProjectVersionStatus["COPYING_IN_PROGRESS"] = "COPYING_IN_PROGRESS";
    ProjectVersionStatus["DELETING"] = "DELETING";
    ProjectVersionStatus["FAILED"] = "FAILED";
    ProjectVersionStatus["RUNNING"] = "RUNNING";
    ProjectVersionStatus["STARTING"] = "STARTING";
    ProjectVersionStatus["STOPPED"] = "STOPPED";
    ProjectVersionStatus["STOPPING"] = "STOPPING";
    ProjectVersionStatus["TRAINING_COMPLETED"] = "TRAINING_COMPLETED";
    ProjectVersionStatus["TRAINING_FAILED"] = "TRAINING_FAILED";
    ProjectVersionStatus["TRAINING_IN_PROGRESS"] = "TRAINING_IN_PROGRESS";
})(ProjectVersionStatus = exports.ProjectVersionStatus || (exports.ProjectVersionStatus = {}));
class InvalidPaginationTokenException extends RekognitionServiceException_1.RekognitionServiceException {
    constructor(opts) {
        super({
            name: "InvalidPaginationTokenException",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidPaginationTokenException";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidPaginationTokenException.prototype);
        this.Message = opts.Message;
        this.Code = opts.Code;
        this.Logref = opts.Logref;
    }
}
exports.InvalidPaginationTokenException = InvalidPaginationTokenException;
var StreamProcessorStatus;
(function (StreamProcessorStatus) {
    StreamProcessorStatus["FAILED"] = "FAILED";
    StreamProcessorStatus["RUNNING"] = "RUNNING";
    StreamProcessorStatus["STARTING"] = "STARTING";
    StreamProcessorStatus["STOPPED"] = "STOPPED";
    StreamProcessorStatus["STOPPING"] = "STOPPING";
    StreamProcessorStatus["UPDATING"] = "UPDATING";
})(StreamProcessorStatus = exports.StreamProcessorStatus || (exports.StreamProcessorStatus = {}));
class ResourceNotReadyException extends RekognitionServiceException_1.RekognitionServiceException {
    constructor(opts) {
        super({
            name: "ResourceNotReadyException",
            $fault: "client",
            ...opts,
        });
        this.name = "ResourceNotReadyException";
        this.$fault = "client";
        Object.setPrototypeOf(this, ResourceNotReadyException.prototype);
        this.Message = opts.Message;
        this.Code = opts.Code;
        this.Logref = opts.Logref;
    }
}
exports.ResourceNotReadyException = ResourceNotReadyException;
class HumanLoopQuotaExceededException extends RekognitionServiceException_1.RekognitionServiceException {
    constructor(opts) {
        super({
            name: "HumanLoopQuotaExceededException",
            $fault: "client",
            ...opts,
        });
        this.name = "HumanLoopQuotaExceededException";
        this.$fault = "client";
        Object.setPrototypeOf(this, HumanLoopQuotaExceededException.prototype);
        this.ResourceType = opts.ResourceType;
        this.QuotaCode = opts.QuotaCode;
        this.ServiceCode = opts.ServiceCode;
        this.Message = opts.Message;
        this.Code = opts.Code;
        this.Logref = opts.Logref;
    }
}
exports.HumanLoopQuotaExceededException = HumanLoopQuotaExceededException;
var TextTypes;
(function (TextTypes) {
    TextTypes["LINE"] = "LINE";
    TextTypes["WORD"] = "WORD";
})(TextTypes = exports.TextTypes || (exports.TextTypes = {}));
var FaceAttributes;
(function (FaceAttributes) {
    FaceAttributes["ALL"] = "ALL";
    FaceAttributes["DEFAULT"] = "DEFAULT";
})(FaceAttributes = exports.FaceAttributes || (exports.FaceAttributes = {}));
var FaceSearchSortBy;
(function (FaceSearchSortBy) {
    FaceSearchSortBy["INDEX"] = "INDEX";
    FaceSearchSortBy["TIMESTAMP"] = "TIMESTAMP";
})(FaceSearchSortBy = exports.FaceSearchSortBy || (exports.FaceSearchSortBy = {}));
var VideoJobStatus;
(function (VideoJobStatus) {
    VideoJobStatus["FAILED"] = "FAILED";
    VideoJobStatus["IN_PROGRESS"] = "IN_PROGRESS";
    VideoJobStatus["SUCCEEDED"] = "SUCCEEDED";
})(VideoJobStatus = exports.VideoJobStatus || (exports.VideoJobStatus = {}));
var VideoColorRange;
(function (VideoColorRange) {
    VideoColorRange["FULL"] = "FULL";
    VideoColorRange["LIMITED"] = "LIMITED";
})(VideoColorRange = exports.VideoColorRange || (exports.VideoColorRange = {}));
var LabelDetectionSortBy;
(function (LabelDetectionSortBy) {
    LabelDetectionSortBy["NAME"] = "NAME";
    LabelDetectionSortBy["TIMESTAMP"] = "TIMESTAMP";
})(LabelDetectionSortBy = exports.LabelDetectionSortBy || (exports.LabelDetectionSortBy = {}));
var PersonTrackingSortBy;
(function (PersonTrackingSortBy) {
    PersonTrackingSortBy["INDEX"] = "INDEX";
    PersonTrackingSortBy["TIMESTAMP"] = "TIMESTAMP";
})(PersonTrackingSortBy = exports.PersonTrackingSortBy || (exports.PersonTrackingSortBy = {}));
var TechnicalCueType;
(function (TechnicalCueType) {
    TechnicalCueType["BLACK_FRAMES"] = "BlackFrames";
    TechnicalCueType["COLOR_BARS"] = "ColorBars";
    TechnicalCueType["CONTENT"] = "Content";
    TechnicalCueType["END_CREDITS"] = "EndCredits";
    TechnicalCueType["OPENING_CREDITS"] = "OpeningCredits";
    TechnicalCueType["SLATE"] = "Slate";
    TechnicalCueType["STUDIO_LOGO"] = "StudioLogo";
})(TechnicalCueType = exports.TechnicalCueType || (exports.TechnicalCueType = {}));
var SegmentType;
(function (SegmentType) {
    SegmentType["SHOT"] = "SHOT";
    SegmentType["TECHNICAL_CUE"] = "TECHNICAL_CUE";
})(SegmentType = exports.SegmentType || (exports.SegmentType = {}));
class IdempotentParameterMismatchException extends RekognitionServiceException_1.RekognitionServiceException {
    constructor(opts) {
        super({
            name: "IdempotentParameterMismatchException",
            $fault: "client",
            ...opts,
        });
        this.name = "IdempotentParameterMismatchException";
        this.$fault = "client";
        Object.setPrototypeOf(this, IdempotentParameterMismatchException.prototype);
        this.Message = opts.Message;
        this.Code = opts.Code;
        this.Logref = opts.Logref;
    }
}
exports.IdempotentParameterMismatchException = IdempotentParameterMismatchException;
var Reason;
(function (Reason) {
    Reason["EXCEEDS_MAX_FACES"] = "EXCEEDS_MAX_FACES";
    Reason["EXTREME_POSE"] = "EXTREME_POSE";
    Reason["LOW_BRIGHTNESS"] = "LOW_BRIGHTNESS";
    Reason["LOW_CONFIDENCE"] = "LOW_CONFIDENCE";
    Reason["LOW_FACE_QUALITY"] = "LOW_FACE_QUALITY";
    Reason["LOW_SHARPNESS"] = "LOW_SHARPNESS";
    Reason["SMALL_BOUNDING_BOX"] = "SMALL_BOUNDING_BOX";
})(Reason = exports.Reason || (exports.Reason = {}));
class MalformedPolicyDocumentException extends RekognitionServiceException_1.RekognitionServiceException {
    constructor(opts) {
        super({
            name: "MalformedPolicyDocumentException",
            $fault: "client",
            ...opts,
        });
        this.name = "MalformedPolicyDocumentException";
        this.$fault = "client";
        Object.setPrototypeOf(this, MalformedPolicyDocumentException.prototype);
        this.Message = opts.Message;
        this.Code = opts.Code;
        this.Logref = opts.Logref;
    }
}
exports.MalformedPolicyDocumentException = MalformedPolicyDocumentException;
class VideoTooLargeException extends RekognitionServiceException_1.RekognitionServiceException {
    constructor(opts) {
        super({
            name: "VideoTooLargeException",
            $fault: "client",
            ...opts,
        });
        this.name = "VideoTooLargeException";
        this.$fault = "client";
        Object.setPrototypeOf(this, VideoTooLargeException.prototype);
        this.Message = opts.Message;
        this.Code = opts.Code;
        this.Logref = opts.Logref;
    }
}
exports.VideoTooLargeException = VideoTooLargeException;
var StreamProcessorParameterToDelete;
(function (StreamProcessorParameterToDelete) {
    StreamProcessorParameterToDelete["ConnectedHomeMinConfidence"] = "ConnectedHomeMinConfidence";
    StreamProcessorParameterToDelete["RegionsOfInterest"] = "RegionsOfInterest";
})(StreamProcessorParameterToDelete = exports.StreamProcessorParameterToDelete || (exports.StreamProcessorParameterToDelete = {}));
const AgeRangeFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.AgeRangeFilterSensitiveLog = AgeRangeFilterSensitiveLog;
const S3ObjectFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.S3ObjectFilterSensitiveLog = S3ObjectFilterSensitiveLog;
const GroundTruthManifestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GroundTruthManifestFilterSensitiveLog = GroundTruthManifestFilterSensitiveLog;
const AssetFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.AssetFilterSensitiveLog = AssetFilterSensitiveLog;
const AudioMetadataFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.AudioMetadataFilterSensitiveLog = AudioMetadataFilterSensitiveLog;
const BeardFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.BeardFilterSensitiveLog = BeardFilterSensitiveLog;
const BlackFrameFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.BlackFrameFilterSensitiveLog = BlackFrameFilterSensitiveLog;
const BoundingBoxFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.BoundingBoxFilterSensitiveLog = BoundingBoxFilterSensitiveLog;
const CoversBodyPartFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CoversBodyPartFilterSensitiveLog = CoversBodyPartFilterSensitiveLog;
const EquipmentDetectionFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.EquipmentDetectionFilterSensitiveLog = EquipmentDetectionFilterSensitiveLog;
const ProtectiveEquipmentBodyPartFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ProtectiveEquipmentBodyPartFilterSensitiveLog = ProtectiveEquipmentBodyPartFilterSensitiveLog;
const EmotionFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.EmotionFilterSensitiveLog = EmotionFilterSensitiveLog;
const LandmarkFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.LandmarkFilterSensitiveLog = LandmarkFilterSensitiveLog;
const PoseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.PoseFilterSensitiveLog = PoseFilterSensitiveLog;
const ImageQualityFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ImageQualityFilterSensitiveLog = ImageQualityFilterSensitiveLog;
const SmileFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.SmileFilterSensitiveLog = SmileFilterSensitiveLog;
const ComparedFaceFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ComparedFaceFilterSensitiveLog = ComparedFaceFilterSensitiveLog;
const KnownGenderFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.KnownGenderFilterSensitiveLog = KnownGenderFilterSensitiveLog;
const CelebrityFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CelebrityFilterSensitiveLog = CelebrityFilterSensitiveLog;
const EyeglassesFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.EyeglassesFilterSensitiveLog = EyeglassesFilterSensitiveLog;
const EyeOpenFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.EyeOpenFilterSensitiveLog = EyeOpenFilterSensitiveLog;
const GenderFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GenderFilterSensitiveLog = GenderFilterSensitiveLog;
const MouthOpenFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.MouthOpenFilterSensitiveLog = MouthOpenFilterSensitiveLog;
const MustacheFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.MustacheFilterSensitiveLog = MustacheFilterSensitiveLog;
const SunglassesFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.SunglassesFilterSensitiveLog = SunglassesFilterSensitiveLog;
const FaceDetailFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.FaceDetailFilterSensitiveLog = FaceDetailFilterSensitiveLog;
const CelebrityDetailFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CelebrityDetailFilterSensitiveLog = CelebrityDetailFilterSensitiveLog;
const CelebrityRecognitionFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CelebrityRecognitionFilterSensitiveLog = CelebrityRecognitionFilterSensitiveLog;
const ComparedSourceImageFaceFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ComparedSourceImageFaceFilterSensitiveLog = ComparedSourceImageFaceFilterSensitiveLog;
const ImageFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ImageFilterSensitiveLog = ImageFilterSensitiveLog;
const CompareFacesRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CompareFacesRequestFilterSensitiveLog = CompareFacesRequestFilterSensitiveLog;
const CompareFacesMatchFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CompareFacesMatchFilterSensitiveLog = CompareFacesMatchFilterSensitiveLog;
const CompareFacesResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CompareFacesResponseFilterSensitiveLog = CompareFacesResponseFilterSensitiveLog;
const ConnectedHomeSettingsFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ConnectedHomeSettingsFilterSensitiveLog = ConnectedHomeSettingsFilterSensitiveLog;
const ConnectedHomeSettingsForUpdateFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ConnectedHomeSettingsForUpdateFilterSensitiveLog = ConnectedHomeSettingsForUpdateFilterSensitiveLog;
const ModerationLabelFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ModerationLabelFilterSensitiveLog = ModerationLabelFilterSensitiveLog;
const ContentModerationDetectionFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ContentModerationDetectionFilterSensitiveLog = ContentModerationDetectionFilterSensitiveLog;
const OutputConfigFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.OutputConfigFilterSensitiveLog = OutputConfigFilterSensitiveLog;
const CopyProjectVersionRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CopyProjectVersionRequestFilterSensitiveLog = CopyProjectVersionRequestFilterSensitiveLog;
const CopyProjectVersionResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CopyProjectVersionResponseFilterSensitiveLog = CopyProjectVersionResponseFilterSensitiveLog;
const CreateCollectionRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CreateCollectionRequestFilterSensitiveLog = CreateCollectionRequestFilterSensitiveLog;
const CreateCollectionResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CreateCollectionResponseFilterSensitiveLog = CreateCollectionResponseFilterSensitiveLog;
const DatasetSourceFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DatasetSourceFilterSensitiveLog = DatasetSourceFilterSensitiveLog;
const CreateDatasetRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CreateDatasetRequestFilterSensitiveLog = CreateDatasetRequestFilterSensitiveLog;
const CreateDatasetResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CreateDatasetResponseFilterSensitiveLog = CreateDatasetResponseFilterSensitiveLog;
const CreateProjectRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CreateProjectRequestFilterSensitiveLog = CreateProjectRequestFilterSensitiveLog;
const CreateProjectResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CreateProjectResponseFilterSensitiveLog = CreateProjectResponseFilterSensitiveLog;
const TestingDataFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.TestingDataFilterSensitiveLog = TestingDataFilterSensitiveLog;
const TrainingDataFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.TrainingDataFilterSensitiveLog = TrainingDataFilterSensitiveLog;
const CreateProjectVersionRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CreateProjectVersionRequestFilterSensitiveLog = CreateProjectVersionRequestFilterSensitiveLog;
const CreateProjectVersionResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CreateProjectVersionResponseFilterSensitiveLog = CreateProjectVersionResponseFilterSensitiveLog;
const StreamProcessorDataSharingPreferenceFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.StreamProcessorDataSharingPreferenceFilterSensitiveLog = StreamProcessorDataSharingPreferenceFilterSensitiveLog;
const KinesisVideoStreamFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.KinesisVideoStreamFilterSensitiveLog = KinesisVideoStreamFilterSensitiveLog;
const StreamProcessorInputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.StreamProcessorInputFilterSensitiveLog = StreamProcessorInputFilterSensitiveLog;
const StreamProcessorNotificationChannelFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.StreamProcessorNotificationChannelFilterSensitiveLog = StreamProcessorNotificationChannelFilterSensitiveLog;
const KinesisDataStreamFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.KinesisDataStreamFilterSensitiveLog = KinesisDataStreamFilterSensitiveLog;
const S3DestinationFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.S3DestinationFilterSensitiveLog = S3DestinationFilterSensitiveLog;
const StreamProcessorOutputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.StreamProcessorOutputFilterSensitiveLog = StreamProcessorOutputFilterSensitiveLog;
const PointFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.PointFilterSensitiveLog = PointFilterSensitiveLog;
const RegionOfInterestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.RegionOfInterestFilterSensitiveLog = RegionOfInterestFilterSensitiveLog;
const FaceSearchSettingsFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.FaceSearchSettingsFilterSensitiveLog = FaceSearchSettingsFilterSensitiveLog;
const StreamProcessorSettingsFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.StreamProcessorSettingsFilterSensitiveLog = StreamProcessorSettingsFilterSensitiveLog;
const CreateStreamProcessorRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CreateStreamProcessorRequestFilterSensitiveLog = CreateStreamProcessorRequestFilterSensitiveLog;
const CreateStreamProcessorResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CreateStreamProcessorResponseFilterSensitiveLog = CreateStreamProcessorResponseFilterSensitiveLog;
const GeometryFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GeometryFilterSensitiveLog = GeometryFilterSensitiveLog;
const CustomLabelFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CustomLabelFilterSensitiveLog = CustomLabelFilterSensitiveLog;
const DatasetChangesFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DatasetChangesFilterSensitiveLog = DatasetChangesFilterSensitiveLog;
const DatasetStatsFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DatasetStatsFilterSensitiveLog = DatasetStatsFilterSensitiveLog;
const DatasetDescriptionFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DatasetDescriptionFilterSensitiveLog = DatasetDescriptionFilterSensitiveLog;
const DatasetLabelStatsFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DatasetLabelStatsFilterSensitiveLog = DatasetLabelStatsFilterSensitiveLog;
const DatasetLabelDescriptionFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DatasetLabelDescriptionFilterSensitiveLog = DatasetLabelDescriptionFilterSensitiveLog;
const DatasetMetadataFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DatasetMetadataFilterSensitiveLog = DatasetMetadataFilterSensitiveLog;
const DeleteCollectionRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteCollectionRequestFilterSensitiveLog = DeleteCollectionRequestFilterSensitiveLog;
const DeleteCollectionResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteCollectionResponseFilterSensitiveLog = DeleteCollectionResponseFilterSensitiveLog;
const DeleteDatasetRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteDatasetRequestFilterSensitiveLog = DeleteDatasetRequestFilterSensitiveLog;
const DeleteDatasetResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteDatasetResponseFilterSensitiveLog = DeleteDatasetResponseFilterSensitiveLog;
const DeleteFacesRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteFacesRequestFilterSensitiveLog = DeleteFacesRequestFilterSensitiveLog;
const DeleteFacesResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteFacesResponseFilterSensitiveLog = DeleteFacesResponseFilterSensitiveLog;
const DeleteProjectRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteProjectRequestFilterSensitiveLog = DeleteProjectRequestFilterSensitiveLog;
const DeleteProjectResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteProjectResponseFilterSensitiveLog = DeleteProjectResponseFilterSensitiveLog;
const DeleteProjectPolicyRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteProjectPolicyRequestFilterSensitiveLog = DeleteProjectPolicyRequestFilterSensitiveLog;
const DeleteProjectPolicyResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteProjectPolicyResponseFilterSensitiveLog = DeleteProjectPolicyResponseFilterSensitiveLog;
const DeleteProjectVersionRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteProjectVersionRequestFilterSensitiveLog = DeleteProjectVersionRequestFilterSensitiveLog;
const DeleteProjectVersionResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteProjectVersionResponseFilterSensitiveLog = DeleteProjectVersionResponseFilterSensitiveLog;
const DeleteStreamProcessorRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteStreamProcessorRequestFilterSensitiveLog = DeleteStreamProcessorRequestFilterSensitiveLog;
const DeleteStreamProcessorResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteStreamProcessorResponseFilterSensitiveLog = DeleteStreamProcessorResponseFilterSensitiveLog;
const DescribeCollectionRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DescribeCollectionRequestFilterSensitiveLog = DescribeCollectionRequestFilterSensitiveLog;
const DescribeCollectionResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DescribeCollectionResponseFilterSensitiveLog = DescribeCollectionResponseFilterSensitiveLog;
const DescribeDatasetRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DescribeDatasetRequestFilterSensitiveLog = DescribeDatasetRequestFilterSensitiveLog;
const DescribeDatasetResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DescribeDatasetResponseFilterSensitiveLog = DescribeDatasetResponseFilterSensitiveLog;
const DescribeProjectsRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DescribeProjectsRequestFilterSensitiveLog = DescribeProjectsRequestFilterSensitiveLog;
const ProjectDescriptionFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ProjectDescriptionFilterSensitiveLog = ProjectDescriptionFilterSensitiveLog;
const DescribeProjectsResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DescribeProjectsResponseFilterSensitiveLog = DescribeProjectsResponseFilterSensitiveLog;
const DescribeProjectVersionsRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DescribeProjectVersionsRequestFilterSensitiveLog = DescribeProjectVersionsRequestFilterSensitiveLog;
const SummaryFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.SummaryFilterSensitiveLog = SummaryFilterSensitiveLog;
const EvaluationResultFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.EvaluationResultFilterSensitiveLog = EvaluationResultFilterSensitiveLog;
const ValidationDataFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ValidationDataFilterSensitiveLog = ValidationDataFilterSensitiveLog;
const TestingDataResultFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.TestingDataResultFilterSensitiveLog = TestingDataResultFilterSensitiveLog;
const TrainingDataResultFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.TrainingDataResultFilterSensitiveLog = TrainingDataResultFilterSensitiveLog;
const ProjectVersionDescriptionFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ProjectVersionDescriptionFilterSensitiveLog = ProjectVersionDescriptionFilterSensitiveLog;
const DescribeProjectVersionsResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DescribeProjectVersionsResponseFilterSensitiveLog = DescribeProjectVersionsResponseFilterSensitiveLog;
const DescribeStreamProcessorRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DescribeStreamProcessorRequestFilterSensitiveLog = DescribeStreamProcessorRequestFilterSensitiveLog;
const DescribeStreamProcessorResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DescribeStreamProcessorResponseFilterSensitiveLog = DescribeStreamProcessorResponseFilterSensitiveLog;
const DetectCustomLabelsRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DetectCustomLabelsRequestFilterSensitiveLog = DetectCustomLabelsRequestFilterSensitiveLog;
const DetectCustomLabelsResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DetectCustomLabelsResponseFilterSensitiveLog = DetectCustomLabelsResponseFilterSensitiveLog;
const DetectFacesRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DetectFacesRequestFilterSensitiveLog = DetectFacesRequestFilterSensitiveLog;
const DetectFacesResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DetectFacesResponseFilterSensitiveLog = DetectFacesResponseFilterSensitiveLog;
const DetectionFilterFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DetectionFilterFilterSensitiveLog = DetectionFilterFilterSensitiveLog;
const DetectLabelsRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DetectLabelsRequestFilterSensitiveLog = DetectLabelsRequestFilterSensitiveLog;
const InstanceFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.InstanceFilterSensitiveLog = InstanceFilterSensitiveLog;
const ParentFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ParentFilterSensitiveLog = ParentFilterSensitiveLog;
const LabelFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.LabelFilterSensitiveLog = LabelFilterSensitiveLog;
const DetectLabelsResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DetectLabelsResponseFilterSensitiveLog = DetectLabelsResponseFilterSensitiveLog;
const HumanLoopDataAttributesFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.HumanLoopDataAttributesFilterSensitiveLog = HumanLoopDataAttributesFilterSensitiveLog;
const HumanLoopConfigFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.HumanLoopConfigFilterSensitiveLog = HumanLoopConfigFilterSensitiveLog;
const DetectModerationLabelsRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DetectModerationLabelsRequestFilterSensitiveLog = DetectModerationLabelsRequestFilterSensitiveLog;
const HumanLoopActivationOutputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.HumanLoopActivationOutputFilterSensitiveLog = HumanLoopActivationOutputFilterSensitiveLog;
const DetectModerationLabelsResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DetectModerationLabelsResponseFilterSensitiveLog = DetectModerationLabelsResponseFilterSensitiveLog;
const ProtectiveEquipmentSummarizationAttributesFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ProtectiveEquipmentSummarizationAttributesFilterSensitiveLog = ProtectiveEquipmentSummarizationAttributesFilterSensitiveLog;
const DetectProtectiveEquipmentRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DetectProtectiveEquipmentRequestFilterSensitiveLog = DetectProtectiveEquipmentRequestFilterSensitiveLog;
const ProtectiveEquipmentPersonFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ProtectiveEquipmentPersonFilterSensitiveLog = ProtectiveEquipmentPersonFilterSensitiveLog;
const ProtectiveEquipmentSummaryFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ProtectiveEquipmentSummaryFilterSensitiveLog = ProtectiveEquipmentSummaryFilterSensitiveLog;
const DetectProtectiveEquipmentResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DetectProtectiveEquipmentResponseFilterSensitiveLog = DetectProtectiveEquipmentResponseFilterSensitiveLog;
const DetectTextFiltersFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DetectTextFiltersFilterSensitiveLog = DetectTextFiltersFilterSensitiveLog;
const DetectTextRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DetectTextRequestFilterSensitiveLog = DetectTextRequestFilterSensitiveLog;
const TextDetectionFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.TextDetectionFilterSensitiveLog = TextDetectionFilterSensitiveLog;
const DetectTextResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DetectTextResponseFilterSensitiveLog = DetectTextResponseFilterSensitiveLog;
const DistributeDatasetFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DistributeDatasetFilterSensitiveLog = DistributeDatasetFilterSensitiveLog;
const DistributeDatasetEntriesRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DistributeDatasetEntriesRequestFilterSensitiveLog = DistributeDatasetEntriesRequestFilterSensitiveLog;
const DistributeDatasetEntriesResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DistributeDatasetEntriesResponseFilterSensitiveLog = DistributeDatasetEntriesResponseFilterSensitiveLog;
const FaceFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.FaceFilterSensitiveLog = FaceFilterSensitiveLog;
const FaceDetectionFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.FaceDetectionFilterSensitiveLog = FaceDetectionFilterSensitiveLog;
const FaceMatchFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.FaceMatchFilterSensitiveLog = FaceMatchFilterSensitiveLog;
const FaceRecordFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.FaceRecordFilterSensitiveLog = FaceRecordFilterSensitiveLog;
const GetCelebrityInfoRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetCelebrityInfoRequestFilterSensitiveLog = GetCelebrityInfoRequestFilterSensitiveLog;
const GetCelebrityInfoResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetCelebrityInfoResponseFilterSensitiveLog = GetCelebrityInfoResponseFilterSensitiveLog;
const GetCelebrityRecognitionRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetCelebrityRecognitionRequestFilterSensitiveLog = GetCelebrityRecognitionRequestFilterSensitiveLog;
const VideoMetadataFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.VideoMetadataFilterSensitiveLog = VideoMetadataFilterSensitiveLog;
const GetCelebrityRecognitionResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetCelebrityRecognitionResponseFilterSensitiveLog = GetCelebrityRecognitionResponseFilterSensitiveLog;
const GetContentModerationRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetContentModerationRequestFilterSensitiveLog = GetContentModerationRequestFilterSensitiveLog;
const GetContentModerationResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetContentModerationResponseFilterSensitiveLog = GetContentModerationResponseFilterSensitiveLog;
const GetFaceDetectionRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetFaceDetectionRequestFilterSensitiveLog = GetFaceDetectionRequestFilterSensitiveLog;
const GetFaceDetectionResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetFaceDetectionResponseFilterSensitiveLog = GetFaceDetectionResponseFilterSensitiveLog;
const GetFaceSearchRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetFaceSearchRequestFilterSensitiveLog = GetFaceSearchRequestFilterSensitiveLog;
const PersonDetailFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.PersonDetailFilterSensitiveLog = PersonDetailFilterSensitiveLog;
const PersonMatchFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.PersonMatchFilterSensitiveLog = PersonMatchFilterSensitiveLog;
const GetFaceSearchResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetFaceSearchResponseFilterSensitiveLog = GetFaceSearchResponseFilterSensitiveLog;
const GetLabelDetectionRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetLabelDetectionRequestFilterSensitiveLog = GetLabelDetectionRequestFilterSensitiveLog;
const LabelDetectionFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.LabelDetectionFilterSensitiveLog = LabelDetectionFilterSensitiveLog;
const GetLabelDetectionResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetLabelDetectionResponseFilterSensitiveLog = GetLabelDetectionResponseFilterSensitiveLog;
const GetPersonTrackingRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetPersonTrackingRequestFilterSensitiveLog = GetPersonTrackingRequestFilterSensitiveLog;
const PersonDetectionFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.PersonDetectionFilterSensitiveLog = PersonDetectionFilterSensitiveLog;
const GetPersonTrackingResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetPersonTrackingResponseFilterSensitiveLog = GetPersonTrackingResponseFilterSensitiveLog;
const GetSegmentDetectionRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetSegmentDetectionRequestFilterSensitiveLog = GetSegmentDetectionRequestFilterSensitiveLog;
const ShotSegmentFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ShotSegmentFilterSensitiveLog = ShotSegmentFilterSensitiveLog;
const TechnicalCueSegmentFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.TechnicalCueSegmentFilterSensitiveLog = TechnicalCueSegmentFilterSensitiveLog;
const SegmentDetectionFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.SegmentDetectionFilterSensitiveLog = SegmentDetectionFilterSensitiveLog;
const SegmentTypeInfoFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.SegmentTypeInfoFilterSensitiveLog = SegmentTypeInfoFilterSensitiveLog;
const GetSegmentDetectionResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetSegmentDetectionResponseFilterSensitiveLog = GetSegmentDetectionResponseFilterSensitiveLog;
const GetTextDetectionRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetTextDetectionRequestFilterSensitiveLog = GetTextDetectionRequestFilterSensitiveLog;
const TextDetectionResultFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.TextDetectionResultFilterSensitiveLog = TextDetectionResultFilterSensitiveLog;
const GetTextDetectionResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetTextDetectionResponseFilterSensitiveLog = GetTextDetectionResponseFilterSensitiveLog;
const IndexFacesRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.IndexFacesRequestFilterSensitiveLog = IndexFacesRequestFilterSensitiveLog;
const UnindexedFaceFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.UnindexedFaceFilterSensitiveLog = UnindexedFaceFilterSensitiveLog;
const IndexFacesResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.IndexFacesResponseFilterSensitiveLog = IndexFacesResponseFilterSensitiveLog;
const KinesisVideoStreamStartSelectorFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.KinesisVideoStreamStartSelectorFilterSensitiveLog = KinesisVideoStreamStartSelectorFilterSensitiveLog;
const ListCollectionsRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ListCollectionsRequestFilterSensitiveLog = ListCollectionsRequestFilterSensitiveLog;
const ListCollectionsResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ListCollectionsResponseFilterSensitiveLog = ListCollectionsResponseFilterSensitiveLog;
const ListDatasetEntriesRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ListDatasetEntriesRequestFilterSensitiveLog = ListDatasetEntriesRequestFilterSensitiveLog;
const ListDatasetEntriesResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ListDatasetEntriesResponseFilterSensitiveLog = ListDatasetEntriesResponseFilterSensitiveLog;
const ListDatasetLabelsRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ListDatasetLabelsRequestFilterSensitiveLog = ListDatasetLabelsRequestFilterSensitiveLog;
const ListDatasetLabelsResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ListDatasetLabelsResponseFilterSensitiveLog = ListDatasetLabelsResponseFilterSensitiveLog;
const ListFacesRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ListFacesRequestFilterSensitiveLog = ListFacesRequestFilterSensitiveLog;
const ListFacesResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ListFacesResponseFilterSensitiveLog = ListFacesResponseFilterSensitiveLog;
const ListProjectPoliciesRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ListProjectPoliciesRequestFilterSensitiveLog = ListProjectPoliciesRequestFilterSensitiveLog;
const ProjectPolicyFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ProjectPolicyFilterSensitiveLog = ProjectPolicyFilterSensitiveLog;
const ListProjectPoliciesResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ListProjectPoliciesResponseFilterSensitiveLog = ListProjectPoliciesResponseFilterSensitiveLog;
const ListStreamProcessorsRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ListStreamProcessorsRequestFilterSensitiveLog = ListStreamProcessorsRequestFilterSensitiveLog;
const StreamProcessorFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.StreamProcessorFilterSensitiveLog = StreamProcessorFilterSensitiveLog;
const ListStreamProcessorsResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ListStreamProcessorsResponseFilterSensitiveLog = ListStreamProcessorsResponseFilterSensitiveLog;
const ListTagsForResourceRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ListTagsForResourceRequestFilterSensitiveLog = ListTagsForResourceRequestFilterSensitiveLog;
const ListTagsForResourceResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ListTagsForResourceResponseFilterSensitiveLog = ListTagsForResourceResponseFilterSensitiveLog;
const NotificationChannelFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.NotificationChannelFilterSensitiveLog = NotificationChannelFilterSensitiveLog;
const PutProjectPolicyRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.PutProjectPolicyRequestFilterSensitiveLog = PutProjectPolicyRequestFilterSensitiveLog;
const PutProjectPolicyResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.PutProjectPolicyResponseFilterSensitiveLog = PutProjectPolicyResponseFilterSensitiveLog;
const RecognizeCelebritiesRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.RecognizeCelebritiesRequestFilterSensitiveLog = RecognizeCelebritiesRequestFilterSensitiveLog;
const RecognizeCelebritiesResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.RecognizeCelebritiesResponseFilterSensitiveLog = RecognizeCelebritiesResponseFilterSensitiveLog;
const SearchFacesRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.SearchFacesRequestFilterSensitiveLog = SearchFacesRequestFilterSensitiveLog;
const SearchFacesResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.SearchFacesResponseFilterSensitiveLog = SearchFacesResponseFilterSensitiveLog;
const SearchFacesByImageRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.SearchFacesByImageRequestFilterSensitiveLog = SearchFacesByImageRequestFilterSensitiveLog;
const SearchFacesByImageResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.SearchFacesByImageResponseFilterSensitiveLog = SearchFacesByImageResponseFilterSensitiveLog;
const VideoFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.VideoFilterSensitiveLog = VideoFilterSensitiveLog;
const StartCelebrityRecognitionRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.StartCelebrityRecognitionRequestFilterSensitiveLog = StartCelebrityRecognitionRequestFilterSensitiveLog;
const StartCelebrityRecognitionResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.StartCelebrityRecognitionResponseFilterSensitiveLog = StartCelebrityRecognitionResponseFilterSensitiveLog;
const StartContentModerationRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.StartContentModerationRequestFilterSensitiveLog = StartContentModerationRequestFilterSensitiveLog;
const StartContentModerationResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.StartContentModerationResponseFilterSensitiveLog = StartContentModerationResponseFilterSensitiveLog;
const StartFaceDetectionRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.StartFaceDetectionRequestFilterSensitiveLog = StartFaceDetectionRequestFilterSensitiveLog;
const StartFaceDetectionResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.StartFaceDetectionResponseFilterSensitiveLog = StartFaceDetectionResponseFilterSensitiveLog;
const StartFaceSearchRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.StartFaceSearchRequestFilterSensitiveLog = StartFaceSearchRequestFilterSensitiveLog;
const StartFaceSearchResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.StartFaceSearchResponseFilterSensitiveLog = StartFaceSearchResponseFilterSensitiveLog;
const StartLabelDetectionRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.StartLabelDetectionRequestFilterSensitiveLog = StartLabelDetectionRequestFilterSensitiveLog;
const StartLabelDetectionResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.StartLabelDetectionResponseFilterSensitiveLog = StartLabelDetectionResponseFilterSensitiveLog;
const StartPersonTrackingRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.StartPersonTrackingRequestFilterSensitiveLog = StartPersonTrackingRequestFilterSensitiveLog;
const StartPersonTrackingResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.StartPersonTrackingResponseFilterSensitiveLog = StartPersonTrackingResponseFilterSensitiveLog;
const StartProjectVersionRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.StartProjectVersionRequestFilterSensitiveLog = StartProjectVersionRequestFilterSensitiveLog;
const StartProjectVersionResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.StartProjectVersionResponseFilterSensitiveLog = StartProjectVersionResponseFilterSensitiveLog;
const StartShotDetectionFilterFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.StartShotDetectionFilterFilterSensitiveLog = StartShotDetectionFilterFilterSensitiveLog;
const StartTechnicalCueDetectionFilterFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.StartTechnicalCueDetectionFilterFilterSensitiveLog = StartTechnicalCueDetectionFilterFilterSensitiveLog;
const StartSegmentDetectionFiltersFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.StartSegmentDetectionFiltersFilterSensitiveLog = StartSegmentDetectionFiltersFilterSensitiveLog;
const StartSegmentDetectionRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.StartSegmentDetectionRequestFilterSensitiveLog = StartSegmentDetectionRequestFilterSensitiveLog;
const StartSegmentDetectionResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.StartSegmentDetectionResponseFilterSensitiveLog = StartSegmentDetectionResponseFilterSensitiveLog;
const StreamProcessingStartSelectorFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.StreamProcessingStartSelectorFilterSensitiveLog = StreamProcessingStartSelectorFilterSensitiveLog;
const StreamProcessingStopSelectorFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.StreamProcessingStopSelectorFilterSensitiveLog = StreamProcessingStopSelectorFilterSensitiveLog;
const StartStreamProcessorRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.StartStreamProcessorRequestFilterSensitiveLog = StartStreamProcessorRequestFilterSensitiveLog;
const StartStreamProcessorResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.StartStreamProcessorResponseFilterSensitiveLog = StartStreamProcessorResponseFilterSensitiveLog;
const StartTextDetectionFiltersFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.StartTextDetectionFiltersFilterSensitiveLog = StartTextDetectionFiltersFilterSensitiveLog;
const StartTextDetectionRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.StartTextDetectionRequestFilterSensitiveLog = StartTextDetectionRequestFilterSensitiveLog;
const StartTextDetectionResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.StartTextDetectionResponseFilterSensitiveLog = StartTextDetectionResponseFilterSensitiveLog;
const StopProjectVersionRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.StopProjectVersionRequestFilterSensitiveLog = StopProjectVersionRequestFilterSensitiveLog;
const StopProjectVersionResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.StopProjectVersionResponseFilterSensitiveLog = StopProjectVersionResponseFilterSensitiveLog;
const StopStreamProcessorRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.StopStreamProcessorRequestFilterSensitiveLog = StopStreamProcessorRequestFilterSensitiveLog;
const StopStreamProcessorResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.StopStreamProcessorResponseFilterSensitiveLog = StopStreamProcessorResponseFilterSensitiveLog;
const TagResourceRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.TagResourceRequestFilterSensitiveLog = TagResourceRequestFilterSensitiveLog;
const TagResourceResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.TagResourceResponseFilterSensitiveLog = TagResourceResponseFilterSensitiveLog;
const UntagResourceRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.UntagResourceRequestFilterSensitiveLog = UntagResourceRequestFilterSensitiveLog;
const UntagResourceResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.UntagResourceResponseFilterSensitiveLog = UntagResourceResponseFilterSensitiveLog;
const UpdateDatasetEntriesRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.UpdateDatasetEntriesRequestFilterSensitiveLog = UpdateDatasetEntriesRequestFilterSensitiveLog;
const UpdateDatasetEntriesResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.UpdateDatasetEntriesResponseFilterSensitiveLog = UpdateDatasetEntriesResponseFilterSensitiveLog;
const StreamProcessorSettingsForUpdateFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.StreamProcessorSettingsForUpdateFilterSensitiveLog = StreamProcessorSettingsForUpdateFilterSensitiveLog;
const UpdateStreamProcessorRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.UpdateStreamProcessorRequestFilterSensitiveLog = UpdateStreamProcessorRequestFilterSensitiveLog;
const UpdateStreamProcessorResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.UpdateStreamProcessorResponseFilterSensitiveLog = UpdateStreamProcessorResponseFilterSensitiveLog;
